tgui/packages/tgui/interfaces/TEG.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author Sovexe (https://github.com/Sovexe)
+ * @license ISC
+ */
+
+import { useBackend } from '../backend';
+import { Box, Chart, LabeledList, Section, Divider } from '../components';
+import { formatPower, formatSiUnit } from '../format';
+import { Window } from '../layouts';
+
+export const TEG = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    output,
+    history,
+    hotCircStatus,
+    hotInletTemp,
+    hotOutletTemp,
+    hotInletPres,
+    hotOutletPres,
+    coldCircStatus,
+    coldInletTemp,
+    coldOutletTemp,
+    coldInletPres,
+    coldOutletPres,
+  } = data;
+  const historyData = history.map((value, i) => [i, value]);
+  const historyMax = Math.max(...history);
+  return (
+    <Window
+      height="520"
+      width="300" >
+      <Window.Content>
+        <Section title="Status">
+          <LabeledList>
+            <LabeledList.Item label="Output History" />
+          </LabeledList>
+          <Chart.Line
+            height="5em"
+            data={historyData}
+            rangeX={[0, historyData.length - 1]}
+            rangeY={[0, historyMax]}
+            strokeColor="rgba(1, 184, 170, 1)"
+            fillColor="rgba(1, 184, 170, 0.25)" />
```

Comment:
Fairly arbitrary colors. They look nice here, but should probably be moved to (and then drawn from) the theme.